meson.build

```
@@ -0,0 +1,471 @@
+            endif
+        endforeach
+        if cc.compiles('#include <magick/api.h>\nColorspaceType colorspace = CMYColorspace;', dependencies: magick_dep)
+            cfg_var.set('HAVE_CMYCOLORSPACE', '1')
+        endif
+        if cc.compiles('#include <magick/api.h>\nColorspaceType colorspace = HCLpColorspace;', dependencies: magick_dep)
+            cfg_var.set('HAVE_HCLPCOLORSPACE', '1')
+        endif
+        # GetImageMagick() takes two args under GM, three under IM
+        if cc.compiles('#include <magick/api.h>\nint main() {(void)GetImageMagick(NULL, 0, NULL);}', dependencies: magick_dep)
+            cfg_var.set('HAVE_GETIMAGEMAGICK3', '1')
+        endif
+    else
+        error('Unsupported libmagick version')
+    endif
+
+    if 'load' in get_option('magick-features')
+        cfg_var.set('ENABLE_MAGICKLOAD', '1')
+    endif
+    if 'save' in get_option('magick-features')
+        cfg_var.set('ENABLE_MAGICKSAVE', '1')
+    endif
+
+    if get_option('magick-module')
+        cfg_var.set('MAGICK_MODULE', '1')
+    endif
+endif
+
+cfitsio_dep = dependency('cfitsio', required: get_option('cfitsio'))
+if cfitsio_dep.found()
+    libvips_deps += cfitsio_dep
+    cfg_var.set('HAVE_CFITSIO', '1')
+endif
+
+# quant package we use
+quantisation_package = disabler()
+
+imagequant_dep = dependency('imagequant', required: get_option('imagequant'))
+if imagequant_dep.found()
+    libvips_deps += imagequant_dep
+    cfg_var.set('HAVE_IMAGEQUANT', '1')
+endif
+
+# only if libimagequant not found
+if not quantisation_package.found()
+    quantizr_dep = dependency('quantizr', required: get_option('quantizr'))
+    if quantizr_dep.found()
+        libvips_deps += quantizr_depquantizr_dep
+        cfg_var.set('HAVE_QUANTIZR', '1')
+        quantisation_package = quantizr_dep
+    endif
+endif
+
+if quantisation_package.found()
+    cgif_dep = dependency('cgif', required: get_option('cgif'))
+    if quantizr_dep.found()
+        libvips_deps += cgif_dep
+        cfg_var.set('HAVE_CGIF', '1')
+    endif
+endif
+
+libexif_dep = dependency('libexif', version: '>=0.6', required: get_option('exif'))
+if libexif_dep.found()
+    libvips_deps += libexif_dep
+    cfg_var.set('HAVE_EXIF', '1')
+    # some libexif packages need include <libexif/poop.h>, some just <poop.h>
+    # how annoying
+    if cc.has_header('exif-data.h', dependencies: libexif_dep)
+        # libexif includes don't need libexif prefix
+        cfg_var.set('UNTAGGED_EXIF', '1')
+    endif
+endif
+
+
+libjpeg_dep = dependency('libjpeg', required: get_option('jpeg'))
+if libjpeg_dep.found()
+    libvips_deps += libjpeg_dep
+    cfg_var.set('HAVE_JPEG', '1')
+    # features like trellis quant are exposed as extension parameters ...
+    # mozjpeg 3.2 and later have #define JPEG_C_PARAM_SUPPORTED, but we must 
+    # work with earlier versions
+    if cc.has_function('jpeg_c_bool_param_supported', prefix: '#include <jpeglib.h>', dependencies: libjpeg_dep)
```

`stdio.h` needs to be included before `jpeglib.h`. See: https://github.com/libjpeg-turbo/libjpeg-turbo/issues/17#issuecomment-139620135. Otherwise, `HAVE_JPEG_EXT_PARAMS` would never be defined when building with MozJPEG.
```suggestion
    if cc.has_function('jpeg_c_bool_param_supported',
                        prefix: '''#include <stdio.h>
                                   #include <jpeglib.h>''',
                        dependencies: libjpeg_dep)
```

---

thanks, fixed